### ISSUE TITLE
Swipe gestures indicators

### ISSUE THREAD

**Issue by spaceisstrange** [minbrowser/min#384]
It'd be nice to have some indicators when the swipe gestures are triggered just like Safari (transitions between the current page and the next one) or Chrome (shows a back or forward arrow as you swipe) does.

I was looking to implement something like this into the browser, but I really don't know how to begin implementing something like this.

**Comment by PalmerAL** [comment]
Having an animation like Safari does would be difficult, since we'd need to take a screenshot of every page in order to show the previous page in the animation (which would likely be fairly slow). However, an arrow like Chrome does shouldn't be too difficult.

If you want to implement this, <a href="https://github.com/minbrowser/min/blob/112bdc2cd4eefae1022cfad82c488238074ee5f3/js/webviews.js#L72-L88">this code</a> runs whenever you swipe. Basically, you would need to show an arrow element here (which could just be placed on top of the webview using position:fixed) and then hide it after a timeout.

**Comment by spaceisstrange** [comment]
Thanks! I'll try to implement it. However, to achieve the same effect as Chrome does the arrow should be appearing at the same time as the swipe is being performed not at the end, so I guess that it should be in the [swipeEvents](https://github.com/minbrowser/min/blob/112bdc2cd4eefae1022cfad82c488238074ee5f3/js/webview/swipeEvents.js) file?

**Comment by PalmerAL** [comment]
Yes. I think the best solution would be to add something <a href="https://github.com/minbrowser/min/blob/112bdc2cd4eefae1022cfad82c488238074ee5f3/js/webview/swipeEvents.js#L51">here</a> like this:

```
if(horizontalMouseMove < -some_number) {
  ipc.sendToHost("showBackArrow");
} else if (horizontalMouseMove > some_number) {
  ipc.sendToHost("showForwardArrow");
}
```

Then you can add another ```bindWebviewIPC``` call in webviews.js:

```
bindWebviewIPC("showBackArrow", function() {
  //show the back arrow element
});
```

Chrome has some kind of effect where the arrow appears and then gets bigger as you swipe more, but we can probably just ignore that and have a fixed animation when the arrow appears.

**Comment by PalmerAL** [comment]
Actually, you would need to check if the arrow has already been shown:

```
if(horizontalMouseMove < -some_number && !hasShownSwipeArrow) {
  ipc.sendToHost("showBackArrow");
  hasShownSwipeArrow = true;
} else if (horizontalMouseMove > some_number && !hasShownSwipeArrow) {
  ipc.sendToHost("showForwardArrow");
  hasShownSwipeArrow = true;
}
```

**Comment by spaceisstrange** [comment]
Great, thank you! I'll try to implement it in my local branch and reach back if I have anything interesting. 

_Sent from my Xiaomi MI 5C using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by code-hunger** [comment]
@revolter https://github.com/minbrowser/min/blob/d08e35dc762bf2fb71eb05ff16db00e984aefe86/js/webviewGestures.js#L1-L7

**Comment by MarkWahlsten** [comment]
It looks like this was implemented, but I don't see the animation (on OSX Mojave).

Visual feedback would be really useful with the keyboard shortcut too. 

Sometimes when my internet connection is slow (or my Command key is a bit faulty), it's hard to tell whether the gesture or keyboard shortcut actually worked. In those cases I often end up swiping, then using the keyboard shortcut, then giving up and clicking the actual button itself anyway (as it's the only way to be sure Min got the message). 

Occasionally I'll just daydream and wait for the page to load for several minutes, until realising the gesture/shortcut didn't work.

**Comment by PalmerAL** [comment]
@MarkWahlsten It stopped working when we switched to BrowserViews a couple years ago; to fix it, we would need to add another view on top that acts as an overlay.

You should get a progress bar after a couple seconds if the new page is loading, but I agree it would be nice to have more immediate feedback; an animation on the back button might work.
